gluten-flink/ut/src/test/java/org/apache/gluten/streaming/api/operators/GlutenStreamFilterTest.java

```diff
@@ -0,0 +1,252 @@
+    rexBuilder = new FlinkRexBuilder(typeFactory);
+  }
+
+  @Override
+  @Test
+  public void testFilter() throws Exception {
+    RowType flinkRowType =
+        RowType.of(
+            new LogicalType[] {
+              new IntType(), new VarCharType(VarCharType.MAX_LENGTH), new IntType()
+            },
+            new String[] {"id", "name", "age"});
+    List<String> inNames = Utils.getNamesFromRowType(flinkRowType);
+
+    RexNode ageFieldRef =
+        rexBuilder.makeInputRef(typeFactory.createSqlType(SqlTypeName.INTEGER), 2);
+    RexNode literal18 =
+        rexBuilder.makeLiteral(
+            new BigDecimal(18), typeFactory.createSqlType(SqlTypeName.INTEGER), false);
+    RexNode filterConditionFlink =
+        rexBuilder.makeCall(SqlStdOperatorTable.GREATER_THAN, ageFieldRef, literal18);
+
+    RexConversionContext conversionContext = new RexConversionContext(inNames);
+    TypedExpr veloxFilterCondition =
+        RexNodeConverter.toTypedExpr(filterConditionFlink, conversionContext);
+
+    io.github.zhztheplayer.velox4j.type.RowType veloxType =
+        (io.github.zhztheplayer.velox4j.type.RowType)
+            org.apache.gluten.util.LogicalTypeConverter.toVLType(flinkRowType);
+
+    PlanNode veloxPlan =
+        new FilterNode(PlanNodeIdGenerator.newId(), List.of(), veloxFilterCondition);
+    TypeInformation<RowData> typeInfo = InternalTypeInfo.of(flinkRowType);
+
+    GlutenSingleInputOperator operator =
+        new GlutenSingleInputOperator(veloxPlan, PlanNodeIdGenerator.newId(), veloxType, veloxType);
+
+    TypeSerializer<RowData> serializer = typeInfo.createSerializer(new SerializerConfigImpl());
+    OneInputStreamOperatorTestHarness<RowData, RowData> harness =
+        new OneInputStreamOperatorTestHarness<>(operator, serializer);
+
+    harness.setup(typeInfo.createSerializer(new SerializerConfigImpl()));
+    harness.open();
+
+    List<RowData> inputData =
+        Arrays.asList(
+            GenericRowData.of(1, StringData.fromString("Alice"), 20),
+            GenericRowData.of(2, StringData.fromString("Bob"), 17),
+            GenericRowData.of(3, StringData.fromString("Charlie"), 25),
+            GenericRowData.of(4, StringData.fromString("David"), 15));
+
+    long timestamp = 0L;
+    for (RowData row : inputData) {
+      harness.processElement(new StreamRecord<>(row, timestamp++));
+    }
+
+    Queue<Object> outputQueue = harness.getOutput();
+    List<RowData> actualOutput = new ArrayList<>();
+    while (!outputQueue.isEmpty()) {
+      Object record = outputQueue.poll();
+      if (record instanceof StreamRecord) {
+        actualOutput.add(((StreamRecord<RowData>) record).getValue());
+      }
+    }
+
+    List<RowData> expectedOutput =
+        Arrays.asList(
+            GenericRowData.of(1, StringData.fromString("Alice"), 20),
+            GenericRowData.of(3, StringData.fromString("Charlie"), 25));
+
+    assertThat(actualOutput).hasSize(expectedOutput.size());
```

Comment:
Maybe, also add tests for LessThan, EqualTo. We can then move the result check to a helper method for reuse.